### PR TITLE
feat: Add metering point periods contracts for electrical heating

### PR DESCRIPTION
**Pull request by BjarkeMeier**
# Description

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        t.StructField("metering_point_id", t.StringType(), not nullable),
+        #
+        #
+        t.StructField("has_electrical_heating", t.BooleanType(), not nullable),
+        #
+        # 2 | 3 | 6 | NULL
+        t.StructField("net_settlement_group", t.IntegerType(), not nullable),
+        #
+        # The number of the month. 0 is January, 11 is December.
+        t.StructField("net_settlement_group_6_scheduled_meter_reading_date", t.IntegerType(), nullable),
```
Comment (defectiveAi) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:18]: Should January be 1, etc?

Comment (defectiveAi) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:18]: net_settlement_group_scheduled_month_meter_reading or net_settlement_group_scheduled_meter_reading_month

**Review by defectiveAi — DISMISSED** [review]

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py (hunk 1 above)
Comment (BjarkeMeier) [reply source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py]: Ok. We can do that.

Comment (BjarkeMeier) [reply source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py]: Ok. Updated. I added a "6" to the name as it only applies to group 6.

File: source/electrical_heating/contracts/electricity_market__electrical_heating/child_metering_point_periods_v1.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+nullable = True
+
+# All child metering points related to electrical heating.
+#
+# The included metering point types are:
+# 'supply_to_grid' 'consumption_from_grid' | 'electrical_heating' | 'net_consumption'
+child_metering_point_periods_v1 = t.StructType(
+    [
+        #
+        # GRSN number
```
Comment (JesperGomes) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/child_metering_point_periods_v1.py:12]:
```suggestion
        # GSRN number
```

Comment (JesperGomes) [reply source/electrical_heating/contracts/electricity_market__electrical_heating/child_metering_point_periods_v1.py:12]: same goes for the others

**Review by JesperGomes — DISMISSED** [review]

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        # 2 | 3 | 4 | 5 | 6 | 99 | NULL
+        t.StructField("net_settlement_group", t.IntegerType(), not nullable),
+        #
+        # The number of the month. 1 is January, 12 is December.
+        # For all but settlement group 6 the month is January.
+        t.StructField(
+            "settlement_month",
+            t.IntegerType(),
+            nullable,
+        ),
```
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:37]: @khatozen, do we need this information? The master data has already been periodized?

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+nullable = True
+
+# Includes all periods of consumption metering points where electrical heating is registered.
+#
+# A period is created whenever any of the following transaction types are registered:
+# - CHANGESUP: Leverandørskift (BRS-001)
+# - ENDSUPPLY: Leveranceophør (BRS-002)
+# - INCCHGSUP: Håndtering af fejlagtigt leverandørskift (BRS-003)
+# - MSTDATSBM: Fremsendelse af stamdata (BRS-006) - Skift af nettoafregningsgrupper
+# - CLSDWNMP: Nedlæggelse af målepunkt (BRS-007)
```
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:12]:
```suggestion
# - LNKCHLDMP: Tilkobling af D15 til parent i nettoafregningsgruppe 2
# - ULNKCHLDMP: Afkobling af D15 af parent i nettoafregningsgruppe 2
# - CLSDWNMP: Nedlæggelse af målepunkt (BRS-007)
```

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            "settlement_month",
+            t.IntegerType(),
+            nullable,
+        ),
+        #
+        # UTC time
+        t.StructField("period_from_date", t.TimestampType(), not nullable),
+        #
+        # UTC time
+        t.StructField("period_to_date", t.TimestampType(), nullable),
```
Comment (defectiveAi) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:45]: Is not easier to work with non-nullable period_to_date?

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+# - MOVEINES: Tilflytning - meldt til elleverandøren (BRS-009)
+# - MOVEOUTES: Fraflytning - meldt til elleverandøren (BRS-010)
+# - INCMOVEAUT: Fejlagtig flytning - Automatisk (BRS-011)
+# - INCMOVEMAN: Fejlagtig flytning - Manuel (BRS-011) HTX
+# - MDCNSEHON: Oprettelse af elvarme (BRS-015) Det bliver til BRS-041 i DH3
+# - MDCNSEHOFF: Fjernelse af elvarme (BRS-015) Det bliver til BRS-041 i DH3
+# - CHGSUPSHRT: Leverandørskift med kort varsel (BRS-043). Findes ikke i DH3
+# - MANCHGSUP: Tvunget leverandørskifte på målepunkt (BRS-044).
+# - MANCOR (HTX): Manuelt korrigering
+consumption_metering_point_periods_v1 = t.StructType(
```
Comment (defectiveAi) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:24]: You no longer need has_electrical_heating?

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import pyspark.sql.types as t
+
+nullable = True
+
+# Includes all periods of consumption metering points where electrical heating is registered.
```
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:5]:
```suggestion
# Includes all periods of consumption metering points where electrical heating is registered.
# Metering point periods where the electrical heating flag is not set must be omitted in the data set.
```

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py (hunk 6 above)
Comment (BjarkeMeier) [reply source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:25]: Correct. I've added this missing doc:
"Metering point periods where the electrical heating flag is not set must be omitted in the data set."

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+# Metering point periods where the electrical heating flag is not set must be omitted in the data set.
```
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:6]:
```suggestion
# Metering point periods where the electrical heating flag is not set must be omitted from the data set.
```

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+consumption_metering_point_periods_v1 = t.StructType(
+    [
+        #
+        # GSRN number
+        t.StructField("metering_point_id", t.StringType(), not nullable),
+        #
+        # 2 | 3 | 4 | 5 | 6 | 99 | NULL
+        t.StructField("net_settlement_group", t.IntegerType(), not nullable),
+        #
+        # The number of the month. 1 is January, 12 is December.
```
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:34]:
```suggestion
        # Settlement month is 1st of January for all consumption with electrical heating except for
        # net settlement group 6, where the date is the scheduled meter reading date.
        # The number of the month. 1 is January, 12 is December.
```

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        # For all but settlement group 6 the month is January.
+        t.StructField(
+            "settlement_month",
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:39]: I've updated the field documentation. Is it better now?

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py (hunk 5 above)
Comment (BjarkeMeier) [reply source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:48]: It hardly matters to us, and this is more correct. I prefer this. Do you prefer an arbitrary future date?

File: source/electrical_heating/contracts/electricity_market__electrical_heating/child_metering_point_periods_v1.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import pyspark.sql.types as t
+
+nullable = True
+
+# All child metering points related to electrical heating.
+#
+# The included metering point types are:
+# 'supply_to_grid' 'consumption_from_grid' | 'electrical_heating' | 'net_consumption'
+child_metering_point_periods_v1 = t.StructType(
```
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/child_metering_point_periods_v1.py:9]: @khatozen, what if the D14 has quarterly resolution?

File: source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py [hunk 12]
```diff
@@ -0,0 +1,50 @@
+import pyspark.sql.types as t
+
+nullable = True
+
+# Includes all periods of consumption metering points where electrical heating is registered.
+# Metering point periods where the electrical heating flag is not set must be omitted from the data set.
```
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/electricity_market__electrical_heating/consumption_metering_point_periods_v1.py:6]: @khatozen, is this correct?

**Review by energinet-ajw — APPROVED** [review]

**Pull request merged**